ffs/scheduler/scheduler.go

```diff
@@ -466,6 +463,12 @@ func (s *Scheduler) execute(ctx context.Context, a astore.Action, job ffs.Job) (
 		return ffs.CidInfo{}, nil, fmt.Errorf("getting current cid info from store: %s", err)
 	}
 
+	if a.ReplacedCid.Defined() {
+		if err := s.Untrack(a.ReplacedCid); err != nil && err != astore.ErrNotFound {
+			return ffs.CidInfo{}, nil, fmt.Errorf("untracking replaced cid: %s", err)
+		}
+	}
```

Comment:
This was moved from `push()` to here (when the Job really executes). This is safer since untracking the replaced Cid should only happen whenever the Job starts executing (if it cancels for some reason, it would be wrong to untrack the replaced cid in advance).